src/main/java/fi/aalto/cs/apluscourses/intellij/actions/ActionUtil.java

```diff
@@ -0,0 +1,43 @@
+package fi.aalto.cs.apluscourses.intellij.actions;
+
+import com.intellij.ide.DataManager;
+import com.intellij.openapi.actionSystem.ActionManager;
+import com.intellij.openapi.actionSystem.ActionPlaces;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.DataContext;
+import java.awt.Component;
+import java.awt.event.ActionListener;
+
+public class ActionUtil {
+
+  /**
+   * Launches {@link AnAction}.
+   *
+   * @param actionId    Id of the action.
+   * @param dataContext A {@link DataContext} for the action.
+   */
+  public static void launch(String actionId, DataContext dataContext) {
+    // https://intellij-support.jetbrains.com/hc/en-us/community/posts/206130119/comments/206169635
+    AnAction action = ActionManager.getInstance().getAction(actionId);
+    AnActionEvent event = AnActionEvent.createFromAnAction(action, null, ActionPlaces.UNKNOWN,
+        dataContext);
+    action.actionPerformed(event);
+  }
+
+  /**
+   * Returns an {@link ActionListener} that launches {@link AnAction} corresponding to the given
+   * id each time {@code actionPerformed()} is called.
+   *
+   * @param actionId Id of the action.
+   * @param source A {@link Component} that gives the {@link DataContext} for the action.
+   * @return An {@link ActionListener}
+   */
+  public static ActionListener onEventLauncher(String actionId, Component source) {
```

Comment:
smart

Reply:
Let's consider the expanded form (without lambda):
```java
  public static ActionListener onEventLauncher(String actionId, Component source) {
    return new ActionListener() {
      @Override
      public void actionPerformed(ActionEvent actionEvent) {
        launch(actionId, DataManager.getInstance().getDataContext(source));
      }
    };
  }
```
So this creates a new `java.awt.event.ActionListener` object that just calls `launch()` function above on its `actionPerformed()` method. (`ActionListener` is used e.g. by many Swing classes as callback interface on certain UI events, like button clicked or such.)

Reply:
I've just probably misunderstood the method's purpose, as expected something related to an `Event`, and in that case, returning a `new Listener` on event launch is a bit confusing. So how then the name `onEventLauncher(...)`, is related to the `Event` then, as it seems to return a `Listener`?

Reply:
Yes, the name is bad. My idea was something like "this method returns 'an object that launches anAction when an event occurs', in other words, it returns an 'on-event launcher'".

How about `getOnEventLauncher()`, `createAutomaticLauncher()` or something alike?

Reply:
I'd go at least with the 1st option

Reply:
I made a compromise: `createOnEventLauncer()`